File: verification/verify_load.py
from playwright.sync_api import sync_playwright

def verify_app_load():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        print("Navigating to app...")
        page.goto("http://localhost:5173")

        print("Waiting for title...")
        page.wait_for_selector("text=Visualisation des émissions de CO2", timeout=10000)

        print("Waiting for data load (Globe sphere)...")
        # .sphere-path is rendered by Globe.jsx
        page.wait_for_selector(".sphere-path", timeout=20000)

        print("Waiting for chart bars...")
        # .bar-rect is rendered by TopCountriesChart.jsx
        page.wait_for_selector(".bar-rect", timeout=20000)

        print("Taking screenshot...")
        page.screenshot(path="verification/verification.png")

        print("Verification complete!")
        browser.close()

if __name__ == "__main__":
    verify_app_load()

File: verification/verify_load.js
import { chromium } from 'playwright';

async function verifyAppLoads() {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  try {
    console.log('Navigating to http://localhost:5173...');
    await page.goto('http://localhost:5173', { timeout: 10000 });

    console.log('Waiting for header...');
    await page.waitForSelector('header', { timeout: 15000 });

    console.log('Waiting for main content...');
    await page.waitForSelector('#main-content', { timeout: 15000 });

    console.log('Waiting for chart group...');
    await page.waitForSelector('.chart-group', { timeout: 15000 });

    await page.screenshot({ path: 'verification/app_loaded.png' });
    console.log('Screenshot saved to verification/app_loaded.png');
    console.log('✅ Verification successful');
  } catch (error) {
    console.error('❌ Verification failed:', error);
    process.exit(1);
  } finally {
    await browser.close();
  }
}

verifyAppLoads();
